src/test/java/org/folio/service/inventory/InventoryItemStatusSyncServiceTest.java

```diff
@@ -0,0 +1,145 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+public class InventoryItemStatusSyncServiceTest {
+  private static final String PO_LINE_ID = UUID.randomUUID().toString();
+  @Mock
+  private InventoryItemManager inventoryItemManager;
+  @Mock
+  private RequestContext requestContext;
+
+  @InjectMocks
+  private InventoryItemStatusSyncService itemStatusSyncService;
+
+  @BeforeEach
+  public void init() {
+    MockitoAnnotations.openMocks(this);
+  }
+
+  @Test
+  void testShouldNotUpdateItemStatusWhenOnOrderItemsNotFound() {
+    //given
+    when(inventoryItemManager.getItemsByPoLineIdsAndStatus(eq(List.of(PO_LINE_ID)), eq(ItemStatus.ON_ORDER.value()), any(RequestContext.class)))
+      .thenReturn(Future.succeededFuture(List.of()));
+    Location location = new Location()
+      .withTenantId("tenant1");
+    List<Location> locations = List.of(location);
+    //when
+    itemStatusSyncService.updateInventoryItemStatus(PO_LINE_ID, locations, ItemStatus.ORDER_CLOSED, requestContext);
+    //then
+    verify(inventoryItemManager, never()).updateItem(any(JsonObject.class), any(RequestContext.class));
+  }
+
+  @Test
+  void testUpdateItemStatus() {
+    //given
+    JsonObject item = new JsonObject()
+      .put(InventoryItemManager.ITEM_PURCHASE_ORDER_LINE_IDENTIFIER, PO_LINE_ID);
+    when(inventoryItemManager.getItemsByPoLineIdsAndStatus(eq(List.of(PO_LINE_ID)), eq(ItemStatus.ON_ORDER.value()), any(RequestContext.class)))
+      .thenReturn(Future.succeededFuture(List.of(item)));
+    when(inventoryItemManager.updateItem(eq(item), any(RequestContext.class))).thenReturn(Future.succeededFuture());
+    Location location = new Location()
+      .withTenantId("tenant1");
+    List<Location> locations = List.of(location);
+    //when
+    itemStatusSyncService.updateInventoryItemStatus(PO_LINE_ID, locations, ItemStatus.ORDER_CLOSED, requestContext);
+    //then
+    verify(inventoryItemManager, times(1)).updateItem(eq(item), any(RequestContext.class));
+    assertEquals(ItemStatus.ORDER_CLOSED, ItemStatus.fromValue(item.getJsonObject("status").getString("name")));
+  }
+
+  @Test
+  void testUpdateMultipleItemStatusesBasedOnLocations() {
+    //given
+    JsonObject item = new JsonObject()
+      .put(InventoryItemManager.ITEM_PURCHASE_ORDER_LINE_IDENTIFIER, PO_LINE_ID);
+    when(inventoryItemManager.getItemsByPoLineIdsAndStatus(eq(List.of(PO_LINE_ID)), eq(ItemStatus.ON_ORDER.value()), any(RequestContext.class)))
+      .thenReturn(Future.succeededFuture(List.of(item)));
+    when(inventoryItemManager.updateItem(eq(item), any(RequestContext.class))).thenReturn(Future.succeededFuture());
+    Location location1 = new Location()
+      .withTenantId("tenant1");
+    Location location2 = new Location()
+      .withTenantId("tenant2");
+    List<Location> locations = List.of(location1, location2);
```

Comment:
You could create a method that can generate Locations and choose whether to populate TenantId or not and use it in other locations. Would be better to avoid copy-pastes.

Or have them created in BeforeEach maybe

Reply:
intoroduced new private method